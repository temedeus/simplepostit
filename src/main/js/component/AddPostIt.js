import React from "react";

/**
 * Single post-it representation written as a functional component.
 * @param {*} postIt
 */
const AddPostIt = props => {
  return (
    <div className="add-postit-wrapper">
      <button onClick={props.onClick} className="add-postit">
        Add
      </button>
    </div>
  );
};

export default AddPostIt;
