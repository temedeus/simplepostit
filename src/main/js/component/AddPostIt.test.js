import React from 'react';
import { shallow } from 'enzyme';
import AddPostIt from "./AddPostIt";

describe('AddPostIt', () => {
    it('should render AddPostIt', async () => {
        const action = jest.fn()
        const addPostIt = shallow(<AddPostIt onClick={action} />)
        expect(addPostIt).toMatchSnapshot()
    });

    it('should be possible to activate button with Spacebar', () => {
        const action = jest.fn()
        const addPostIt = shallow(<AddPostIt onClick={action} />)
        addPostIt
            .find('button')
            .simulate('click')

        expect(action).toHaveBeenCalled()
        addPostIt.unmount()
    })
})
