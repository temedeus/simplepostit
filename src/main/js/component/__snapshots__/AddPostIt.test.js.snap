// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPostIt should render AddPostIt 1`] = `
<div
  className="add-postit-wrapper"
>
  <button
    className="add-postit"
    onClick={[MockFunction]}
  >
    Add
  </button>
</div>
`;
